### ISSUE TITLE
Evaluation on 3d images after training 2d planes. 

### ISSUE THREAD

**Issue by rz0012** [MouseLand/cellpose#163]
I really appreciate your work on CellPose. 
I am working on project related to CellPose and I am currently trying to utilize your approach to train 3d images for producing and predicting cytoplasm masks. I have read that the model trains on 2d planes for better results on 3d images. I have split 3d image into z number of xy planes and trained upon that. 

there are 150 3d grayscaled images for training, each includes 59 xy planes, total 2d planes = 8850
the model is training on these 8850 2d planes with respect to their assigned segmentations.  
I am evaluating the trained model using this:

``` 
#train - train contains 8850 2d planes with just raw images.  
masks, flows, styles = model.eval(train, channels=[0,0], do_3D=True, min_size = -1)

#GT_files.p is ground truth for 8850 images
train_ground_truth = np.array(pickle.load(open(r'./Fluo-N3DH-SIM+/GT_files.p', 'rb')))

pickle.dump(masks,open(os.path.join(r'./Fluo-N3DH-SIM+','predicted_masks.p'), 'wb'))

#probability checking, I am using really low threshold for checking purpose
ap,tp,fp,fn = metrics.average_precision(train_ground_truth, masks, threshold=[0.05, 0.1, 0.2])
``` 

The results produces 0 true positives, but many false positives and false negatives.
Is there anything that I am doing wrong in this part?

thank you for the help in advance.

**Comment by carsen-stringer** [comment]
How are you training the model? Are you training it from scratch? Have you tried including XZ and YZ slices in your training? If XZ and YZ look very different from XY then I recommend using stitching instead of the full 3D model (see here: https://cellpose.readthedocs.io/en/latest/settings.html#d-settings)

**Comment by rz0012** [comment]
I understand. Thank you for sharing this with me. It did not catch my eye earlier. You are right, XZ and YZ do look different than XY, very different in some cases. I will try to use this stitching algorithm. Thanks again.
